### PR TITLE
Introduce the logic to load the OTX1.x classification models

### PR DESCRIPTION
**Pull request by sungmanc**
### Summary

Some minor changes could happen, however, I'll handle this at the following PR.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/core/model/module/base.py [hunk 1]
```diff
@@ -135,12 +136,32 @@ def state_dict(self) -> dict[str, Any]:
...
-    def load_state_dict(self, state_dict: dict[str, Any], *args, **kwargs) -> None:
+    def _load_from_prev_otx_ckpt(self, state_dict: dict[str, Any]) -> dict[str, Any]:
+        """Attach the model.model prefix to load without problem."""
+        msg = "Trying to load the model checkpoint created by OTX 1.X. it will be converted to OTX2.0 format."
+        warnings.warn(msg, stacklevel=1)
+        for key in list(state_dict.keys()):
+            value = state_dict.pop(key)
+            new_key = "model.model." + key
+            state_dict[new_key] = value
+        return state_dict
```
Comment (vinnamkim) [inline src/otx/core/model/module/base.py]: Is really this function able to be used in common broadly? If not, please consider converting this to an abstract function. I'm feeling that this PR introduces so many complicate code levels compared to what we really want. It needs to downweighted.
```suggestion
    def _load_from_otx_1_x_ckpt(self, state_dict: dict[str, Any]) -> dict[str, Any]:
        """Attach the model.model prefix to load without problem."""
        msg = "Trying to load the model checkpoint created by OTX 1.X. But, there is no model checkpoint conversion implementation for this model."
        raise NotImplementError(msg)
```

p.s. Consider renaming it to `_load_from_otx_1_x_ckpt` to make it more specific.

File: src/otx/algo/classification/mobilenet_v3_large.py [hunk 2]
```diff
@@ -20,19 +43,29 @@ def __init__(self, num_classes: int, num_multiclass_heads: int, num_multilabel_c
         config.head.num_multilabel_classes = num_multilabel_classes
         super().__init__(num_classes=num_classes, config=config)
 
+    def convert_previous_otx_ckpt(self, state_dict, add_prefix: str=""):
+        """Convert the previous OTX ckpt according to OTX2.0."""
+        return self.convert_previous_otx_ckpt_impl(state_dict, "hlabel", add_prefix)
```
Comment (vinnamkim) [inline src/otx/algo/classification/mobilenet_v3_large.py]:
```suggestion
    def _load_from_otx_1_x_ckpt(self, state_dict):
        """Convert the previous OTX ckpt according to OTX2.0."""
        for key in list(state_dict.keys()):
            val = state_dict.pop(key)
            if key.startswith("classifier."):
                if "4" in key:
                    key = "head." + key.replace("4", "3")
                    if label_type == "multilabel":
                        val = val.t()
                else:
                    key = "head." + key
            elif key.startswith("act"):
                key = "head." + key
            elif not key.startswith("backbone."):
                key = "backbone." + key
            state_dict[add_prefix + key] = val
        return state_dict 
```

File: src/otx/algo/classification/efficientnet_b0.py [hunk 3]
```diff
@@ -10,8 +11,27 @@
...
+            elif key.startswith("output."):
+                key = key.replace("output", "head")
+                if not label_type == "hlabel":
+                    key = key.replace("asl", "fc")
+                val = val.t()
+            state_dict[add_prefix + key] = val
+        return state_dict 
 
-class EfficientNetB0ForHLabelCls(MMPretrainHlabelClsModel):
+class EfficientNetB0ForHLabelCls(EfficientNetB0LoadWeightMixin, MMPretrainHlabelClsModel):
```
Comment (vinnamkim) [inline src/otx/algo/classification/efficientnet_b0.py]: It seems too much to introduce mixin into this. I still think that the complexity by adding this is not necessary to solve our problem.
```suggestion
class EfficientNetB0ForHLabelCls(MMPretrainHlabelClsModel):
```

**Comment by sungmanc** [comment]
@vinnamkim , thanks for giving your opinion. I also want to hear the opinions from dev team about introducing the Mixin class. I'll reflect your reviews after previous PR is merged.

File: src/otx/core/model/module/base.py [hunk 4]
```diff
@@ -156,7 +151,7 @@ def load_state_dict(self, ckpt: dict[str, Any], *args, **kwargs) -> None:
         """
         if is_ckpt_from_otx_v1(ckpt):
             model_state_dict = ckpt["model"]["state_dict"]
-            state_dict = self._load_from_prev_otx_ckpt(model_state_dict)
+            state_dict = self._load_from_otx_v1_ckpt(model_state_dict)
```
Comment (vinnamkim) [inline src/otx/core/model/module/base.py]: Can we just call `self.model`'s function and return here?
```suggestion
            state_dict = self.model.load_from_otx_v1_ckpt(model_state_dict)
            return
```

File: src/otx/core/model/module/base.py (hunk 1 above)
Comment (sungmanc) [reply src/otx/core/model/module/base.py]: Done, https://github.com/openvinotoolkit/training_extensions/pull/2828/commits/7fedf31302bc0b4143d1e32dd99ff70237978cf7

File: src/otx/algo/classification/mobilenet_v3_large.py (hunk 2 above)
Comment (sungmanc) [reply src/otx/algo/classification/mobilenet_v3_large.py]: Changed the design.

File: src/otx/algo/classification/efficientnet_b0.py (hunk 3 above)
Comment (sungmanc) [reply src/otx/algo/classification/efficientnet_b0.py]: Changed.

File: src/otx/core/model/module/base.py (hunk 4 above)
Comment (sungmanc) [reply src/otx/core/model/module/base.py]: Changed.

**Comment by sungmanc** [comment]
@vinnamkim ,  I marked your comments as resolved since we've already discussed about the design.

**Review by vinnamkim — APPROVED** [review]
LGTM.

**Review by harimkang — APPROVED** [review]

**Review by jaegukhyun — APPROVED** [review]

**Pull request merged**